.github/workflows/deploy.yml

```diff
@@ -0,0 +1,213 @@
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  workflow_call:
+    inputs:
+      environment:
+        required: true
+        type: string
+      image_location:
+        description: "Location of the image to deploy."
+        type: string
+        required: false
+      slack_alert_on_failure:
```

Comment:
Really nitpicky 🤏 but could this be called `alert_slack_on_failure` so it lines up with `notify_slack_on_deployment` below please?